crates/biome_js_semantic/src/events.rs

```diff
@@ -719,13 +720,35 @@ impl SemanticEventExtractor {
                 self.pop_scope(node.text_range());
             }
             _ => {
+                if let Some(node) = JsExportDefaultExpressionClause::cast_ref(node) {
+                    self.leave_export_default_expression(&node);
+                }
                 if let Some(node) = AnyTsType::cast_ref(node) {
                     self.leave_any_type(&node);
                 }
             }
         }
     }
 
+    /// This function inspects the [JsReferenceIdentifier] that are exported by the node [JsExportDefaultExpressionClause].
+    ///
+    /// If these references belong to a binding, they are marked as exported.
+    fn leave_export_default_expression(&mut self, node: &JsExportDefaultExpressionClause) {
+        for node in node.syntax().descendants().skip(1) {
+            if let Some(identifier) = JsReferenceIdentifier::cast_ref(&node) {
```

Comment:
I think this code will add more exports than expected. For instance, in the following code, `X` and `Y` will be reported as exported:

```js
export default function f() {
    retun X + Y;
}
```